benchmarks/units.py

```diff
@@ -86,3 +104,64 @@ def time_quantity_times_quantity():
 
 def time_quantity_ufunc_sin():
     np.sin(q2)
+
+
+class TimeQuantityOpSmallArray:
+    """
+    Operator benchmarks from https://github.com/astropy/astropy/issues/7546
+    for a small Numpy array.
+    """
+    def setup(self):
+        data = np.array([1., 2., 3.])
+        self.data = data * u.g
+        self.out = data.copy()
+
+        # A different but dimensionally compatible unit
+        self.data2 = 0.001 * data * u.kg
```

Comment:
Note: In the paper, these benchmarks were also done for `data2` having same unit as `data1`, but I did not include it here. My reasoning is that `data2` having a different unit is a worse case scenario, hence it is unnecessary to include the other one. Please correct me if I am wrong.

Reply:
I included both since it triggers different code paths (and in particular a possible early return if the units are identical).

Reply:
@mhvk , what are your thoughts here and for the comment below this one? Do we need to benchmark all possible routes?

Reply:
I think it is probably good to include the same-unit case - we do indeed try to make that faster, so if it suddenly becomes a lot slower, it would be good to have warning.